### PR TITLE
Drop support for the old GitHub layout

### PR DESCRIPTION
**Pull request by yakov116**
As per https://github.com/sindresorhus/refined-github/pull/3946#discussion_r571110805

Tested:
- [x] `clone-branch`
- [x] `repo-age`
- [x] `bugs-tab`
- [x] `download-folder-button`
- [x] `split-issue-pr-search-results`
- [x] `releases-tab`
- [x] `toggle-files-button`
- [x] `show-open-prs-of-forks`
- [x] `tags-on-commits-list`
- [x] `toggle-files-button`
- [x] `user-profile-follower-badge`
- [x] `set-default-repositories-type-to-sources`
- [x] `pull-request-hotkey`
- [x] `more-dropdown`
- [x] `mark-merge-commits-in-list`
- [x] `follow-file-renames`
- [x] `default-branch-button`
- [x] `comments-time-machine-links`
- [x] `ci-link`

### CODE REVIEW

**Comment by busches** [comment]
This will break all of these features on GHE 2.21, which is still supported until June. I would rather wait until then to remove this code.

**Comment by fregante** [comment]
Maybe we can reopen this PR in June

**Comment by yakov116** [comment]
> Maybe we can reopen this PR in June

Let me finish it then I will close

**Comment by yakov116** [comment]
@fregante please don't review. This went through many rebases, I will mark it ready to review next week sometime.

**Review by fregante — APPROVED** [review]
The code good so far except I haven’t checked releases-tab and the attribute usage in follow-file-renames

**Comment by yakov116** [comment]
> follow-file-renames

The change is that until now we are getting the short hash of the commit. With this change we get the entire hash.

Release-tab is messed up. I will redo it.

File: source/features/releases-tab.tsx [hunk 1]
```diff
@@ -60,78 +54,48 @@ async function init(): Promise<false | void> {
...
-			createDropdownItem('Releases', buildRepoURL('releases'), {
-				'data-menu-item': 'rgh-releases-item'
-			})
-		);
-
-		return;
-	}
+	await elementReady('.UnderlineNav-body');
 
+	const repoNavigationBar = select('.js-responsive-underlinenav .UnderlineNav-body')!;
```
Comment (fregante) [inline source/features/releases-tab.tsx:59]: Is this selecting the same thing elementReady is selecting? Then it’s duplicate

File: source/features/releases-tab.tsx [hunk 2]
```diff
@@ -60,78 +54,46 @@ async function init(): Promise<false | void> {
...
-					<span data-content="Releases">Releases</span>
-					{count && <span className="Counter" title={count > 999 ? String(count) : ''}>{abbreviateNumber(count)}</span>}
-				</a>
-			</li>
-		);
-		repoNavigationBar.append(releasesTab);
-
-		// This re-triggers the overflow listener forcing it to also hide this tab if necessary #3347
-		repoNavigationBar.replaceWith(repoNavigationBar);
+	const repoNavigationBar = (await elementReady('.js-responsive-underlinenav .UnderlineNav-body'))!;
```
Comment (yakov116) [inline source/features/releases-tab.tsx:57]: The extra parentheses are for the non-null check

File: source/features/more-dropdown.tsx [hunk 3]
```diff
@@ -54,72 +36,26 @@ function onlyShowInDropdown(id: string): void {
...
-		<a href={compareUrl} className="rgh-reponav-more dropdown-item">
-			<DiffIcon/> Compare
-		</a>,
-
-		pageDetect.isEnterprise() ? '' : (
-			<a href={dependenciesUrl} className="rgh-reponav-more dropdown-item">
-				<PackageIcon/> Dependencies
-			</a>
-		),
+	select('.js-responsive-underlinenav .UnderlineNav-body')!.parentElement!.classList.add('rgh-has-more-dropdown');
```
Comment (yakov116) [inline source/features/more-dropdown.tsx:47]: Same duplicate selector. Will fix

**Comment by busches** [comment]
@yakov116 let me know when this is ready and I'll do a quick test on GHE.

**Comment by yakov116** [comment]
> @yakov116 let me know when this is ready and I'll do a quick test on GHE.

its ready to test

**Comment by fregante** [comment]
@yakov116 Is there any meta that you're already took care of but is still _unchecked_ here? https://github.com/sindresorhus/refined-github/issues/4008#issuecomment-812843149

**Review by fregante — APPROVED** [review]
Untested but code looks good 💯

**Comment by yakov116** [comment]
> @yakov116 Is there any meta that you're already took care of but is still _unchecked_ here? [#4008 (comment)](https://github.com/sindresorhus/refined-github/issues/4008#issuecomment-812843149)

No

**Review by busches — APPROVED** [review]

**Comment by yakov116** [comment]
@fregante do we plan on doing a release before the first? If not this can be merged.

**Comment by fregante** [comment]
No, we're waiting.

It's up to you if you want to merge this now or give them an extra week to upgrade and then merge the PR after the 21.6.1 release

**Comment by yakov116** [comment]
> It's up to you if you want to merge this now or after the 21.6.1 release, to give them an extra week of support.

As long as we don't have any conflicts, I don't care leaving it open.

File: source/features/comments-time-machine-links.tsx [hunk 4]
```diff
@@ -91,10 +91,7 @@ async function showTimemachineBar(): Promise<void | false> {
 			return false;
 		}
 
-		const lastCommitDate = await elementReady([
-			'.repository-content .Box.Box--condensed relative-time',
-			'[itemprop="dateModified"] relative-time' // "Repository refresh" layout
-		].join(), {waitForChildren: false});
+		const lastCommitDate = await elementReady('[itemprop="dateModified"] relative-time', {waitForChildren: false});
```
Comment (yakov116) [inline source/features/comments-time-machine-links.tsx:94]: I messed this one up. I deleted the wrong one

Comment (fregante) [reply source/features/comments-time-machine-links.tsx:94]: I see. I was seeing the bar because in Safari is still on 21.6.1

Comment (yakov116) [reply source/features/comments-time-machine-links.tsx:94]: I did attempt to fix it in the last PR... 

Whatever..

**Pull request merged**